WalletWasabi.Gui/Behaviors/CommandOnFirstVisible.cs

```diff
@@ -0,0 +1,39 @@
+using Avalonia;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using ReactiveUI;
+using System;
+using System.Collections.Generic;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.Gui.Behaviors
+{
+	internal class CommandOnFirstVisible : CommandBasedBehavior<InputElement>
+	{
+		private CompositeDisposable Disposables { get; set; }
+
+		protected override void OnAttached()
+		{
+			Disposables = new CompositeDisposable();
+
+			base.OnAttached();
+
+			Observable.FromEventPattern(AssociatedObject, nameof(AssociatedObject.AttachedToVisualTree))
+			.Take(1) // Only on first appearance.
+			.Subscribe(_ =>
+			{
+				ExecuteCommand();
```

Comment:
@danwalmsley at this moment the control still not visible for the user. If I start some CPU intense operation it still cannot be seen until it ends. Instead of this I should wait until the control is "rendered" then start the operation. 

I tried `IsEffectiveVisible` with a while loop and `Task.Delay` but not worked. What I do now it to put `Task.Delay` here:

https://github.com/zkSNACKs/WalletWasabi/pull/1398/files#diff-8ef0b233124cda8dfcef1707a9d8df4aR362

so that way the control is likely to be visible when I start the operation. 

Do you have any idea how to avoid this solution which is working but obviously a hack?